pkgs/by-name/ig/igir/package.nix

```
@@ -2,46 +2,47 @@
   lib,
   buildNpmPackage,
   fetchFromGitHub,
-
-  # for patching bundled 7z binary from the 7zip-bin node module
-  # at lib/node_modules/igir/node_modules/7zip-bin/linux/x64/7za
-  autoPatchelfHook,
   stdenv,
+  SDL2,
+  autoPatchelfHook,
 }:
-
 buildNpmPackage rec {
   pname = "igir";
-  version = "2.11.0";
+  version = "3.0.1";
 
   src = fetchFromGitHub {
     owner = "emmercm";
     repo = "igir";
     rev = "v${version}";
-    hash = "sha256-NG0ZP8LOm7fZVecErTuLOfbp1yvXwHnwPkWTBzUJXWE=";
+    hash = "sha256-QPUnvirufY3UiSFVxX3xCmpAlzPKg5JxwAnVefdepqU=";
   };
 
-  npmDepsHash = "sha256-ADIEzr6PkGaJz27GKSVyTsrbz5zbud7BUb+OXPtP1Vo=";
+  npmDepsHash = "sha256-hJzgFWUJnEQu+EGD4itv04IQl7Rwm8CYeFELJ9aoTMY=";
 
   # I have no clue why I have to do this
   postPatch = ''
     patchShebangs scripts/update-readme-help.sh
   '';
 
-  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [
+    (lib.getLib stdenv.cc.cc)
+    SDL2
```

I think these two dependencies are also required:

```suggestion
    SDL2
    libuv
    lz4
```

---

Help me out here - why do you think those two additions are required?

---

Because the build failed without these for me (I'm on x86_64-Linux):

```
       > auto-patchelf: 2 dependencies could not be satisfied
       > error: auto-patchelf could not satisfy dependency libuv.so.1 wanted by /nix/store/4gfk8cs4gaiz19dsw7wsbwbyjarmrz59-igir-3.0.1/lib/node_modules/igir/node_modules/@emmercm/maxcso-linux-x64/dist/maxcso
       > error: auto-patchelf could not satisfy dependency liblz4.so.1 wanted by /nix/store/4gfk8cs4gaiz19dsw7wsbwbyjarmrz59-igir-3.0.1/lib/node_modules/igir/node_modules/@emmercm/maxcso-linux-x64/dist/maxcso
       > auto-patchelf failed to find all the required dependencies.
       > Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.
       For full logs, run 'nix log /nix/store/03jy5h6r00zjffq29xgn2qy5dksp3mjs-igir-3.0.1.drv'.
```

---

Interesting. It does not so that on aarch64. Will add those dependencies then.

---

We can add them just for Linux if it works fine for Darwin:

```nix
buildInputs =
  [
    (lib.getLib stdenv.cc.cc)
    SDL2
  ]
  ++ lib.optionals stdenv.hostPlatform.isLinux [
    libuv
    lz4
  ];
```